ssc/Pos/Ssc/Mode.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DataKinds #-}
+
+module Pos.Ssc.Mode
+       ( SscMode
+       ) where
+
+import           Universum
+
+import           Control.Monad.Catch         (MonadMask)
+import qualified Ether
+import           Mockable                    (Mockable, ThreadId)
+import qualified Mockable                    as M
+import           System.Wlog                 (WithLogger)
+
+import           Pos.Communication.PeerState (WithPeerState)
+import           Pos.Core                    (MonadPrimaryKey)
+import           Pos.DB.Class                (MonadDBCore, MonadDBPure)
+import           Pos.Discovery               (MonadDiscovery)
+import           Pos.Lrc.Context             (LrcContext)
+import           Pos.Recovery.Info           (MonadRecoveryInfo)
+import           Pos.Reporting               (MonadReportingMem)
+import           Pos.Security.Params         (SecurityParams)
+import           Pos.Shutdown                (MonadShutdownMem)
+import           Pos.Slotting                (MonadSlots)
+import           Pos.Ssc.Class.Types         (MonadSscContext)
+import           Pos.Ssc.Extra               (MonadSscMem)
+
+-- | Mode used for all SSC listeners, workers, and the like.
+type SscMode ssc m
+    = ( WithLogger m
+      , MonadIO m
+      , MonadMask m
+      , MonadSlots m
+      , Mockable M.CurrentTime m      -- TODO: if I use Mockables, GHC fails
```

Comment:
It's a bug that is probably fixed in 8.2, and even if it's not it could likely be solved by tweaking some constraint-solving flags (which I didn't want to do).